File: GO/Structs/teste1.go
package main

import "fmt"

type User struct {
    Nome  string
    Idade int
}

func (u User) Saudar() string {
    return fmt.Sprintf("Oi, sou %s!", u.Nome)
}

func (u *User) Aniversario() {
    u.Idade++
    fmt.Printf("Agora tem %d anos!\n", u.Idade)
}

func main() {
    user := User{Nome: "Teste", Idade: 42}
    fmt.Println(user.Saudar())
    user.Aniversario()
}

File: Rust/Structs/teste1.rs
struct User {
    nome: String,
    idade: u32,
}

impl User {
    fn novo(nome: &str, idade: u32) -> User {
        User {
            nome: String::from(nome),
            idade,
        }
    }

    fn saudar(&self) -> String {
        format!("Oi, sou {} e tenho {} anos!", self.nome, self.idade)
    }
}

fn main() {
    let user = User::novo("Teste", 42);
    println!("{}", user.saudar());
}